src/Symfony/Component/Console/Input/InputDefinition.php

```diff
@@ -100,22 +100,22 @@ public function addArgument(InputArgument $argument)
             throw new LogicException(sprintf('An argument with name "%s" already exists.', $argument->getName()));
         }
 
-        if ($this->hasAnArrayArgument) {
-            throw new LogicException('Cannot add an argument after an array argument.');
+        if (null !== $this->lastArrayArgument) {
+            throw new LogicException(sprintf('Cannot add a required argument "%s" after an optional one "%s".', $argument->getName(), $this->lastArrayArgument->getName()));
```

Comment:
this is the wrong error message. This error happens when adding an argument after an array argument, not when adding a required argument after an optional one (which is the next check).
I suspect a bad copy-paste.

Reply:
Yeah, I think it was a copy past.